root/usr/bin/generate-proxy-cert

```diff
@@ -10,8 +10,14 @@ exportKCL
 
 DIR=/etc/foreman-proxy
 DAYS=${KCL_FDI_PROXY_CERT_DAYS:-999}
-
-COMMON_NAME=$(nmcli -t -f IP4.ADDRESS con show primary 2>/dev/null | cut -f2 -d: | cut -f1 -d/)
+WAIT=${KCL_FDI_NMWAIT:-120}
+SECONDS=0
+while (( SECONDS < 60 )); do
+  sleep 1
+  COMMON_NAME=$(nmcli -w $WAIT -t -f IP4.ADDRESS con show primary 2>/dev/null | cut -f2 -d: | cut -f1 -d/)
+  [ ! -z "$COMMON_NAME" ] && break
+  logger "Waiting for IP4 address to generate SSL cert ($SECONDS)"
+done
 
 # Don't fail when IP address was not provided (HTTP can be still used).
```

Comment:
Before proxy can start, it needs SSL cert and it needs IP address for Common Name in order it to be valid. That's why it needs to wait. There is not reasonable way to detect when IP address was assigned other than this.